sympy/functions/combinatorial/sequences.py

```diff
@@ -0,0 +1,375 @@
+
+        >>> from sympy.functions.combinatorial.sequences import andre_sequence
+        >>> andre_sequence(0)
+        []
+
+        >>> a = andre_sequence(12)
+        >>> print(a)  # numbers of alternating permutations
+        [1, 1, 1, 2, 5, 16, 61, 272, 1385, 7936, 50521, 353792]
+
+        >>> print(a[::2])  # secant numbers
+        [1, 1, 5, 61, 1385, 50521]
+
+        >>> print(a[1::2])  # tangent numbers
+        [1, 2, 16, 272, 7936, 353792]
+
+        References
+        ==========
+
+        .. [1] Ludwig Seidel, Ueber eine einfache Entstehungsweise der
+               Bernoulli'schen Zahlen und einiger verwandten Reihen.
+        .. [2] Peter Luschny, The Seidel transform. http://goo.gl/IxYZl0
+        .. [3] OEIS A000111, Euler up/down numbers. http://oeis.org/A000111
+        .. [4] OEIS A094503, Andre polynomials. http://oeis.org/A094503
+    """
+
+    @classmethod
+    def eval(cls, size):
+        s = sympify(size)
+        if not s.is_Integer or s <= Integer(0):
+            return []
+
+        k, e, R = 0, 1, []
+        A = {-1: 0, 0: 1}
+        for i in range(size):
+            ak = 0
+            A[k + e] = 0
+            e = -e
+            for _ in range(i + 1):
+                ak += A[k]
+                A[k] = ak
+                k += e
+            R.append(ak)
+        return R
+
+
+class euler_sequence(Function):
+    """
+        Input
+        =====
+
+        - size -- positive integer
+
+        Output
+        ======
+
+        If 0 < size then return the list of nonzero secant numbers
+        which is also the list of the even unsigned Euler numbers
+
+            [E0, E2, E4, ..., E{2size-2}]
+
+        else return the empty list [].
+
+        The list is defined as the initial segment of OEIS A000364.
+
+        The Euler numbers can be expressed with Bernoulli polynomials:
+
+        E(n) = (2**(1 + 2*n)/(n + 1)) * (B(n + 1, 3/4) - B(n + 1, 1/4))
+
+        Is equivalent to [abs(euler(n)) for n in range(0,2*size,2)].
+        Is equivalent to andre_sequence(2*size)[::2].
+
+        Examples
+        ========
+
+        >>> from sympy.functions.combinatorial.sequences import euler_sequence
+        >>> euler_sequence(0)
+        []
+
+        >>> euler_sequence(9)
+        [1, 1, 5, 61, 1385, 50521, 2702765, 199360981, 19391512145]
+
+        References
+        ==========
+
+        .. [1] OEIS A000364, Euler secant numbers.
+               https://oeis.org/A000364
+
+        See Also
+        ========
+
+        - Andre numbers, OEIS A000111
+        - Tangent numbers, OEIS A000182
+    """
+
+    @classmethod
+    def eval(cls, size):
+        s = sympify(size)
+        if not s.is_Integer or s <= Integer(0):
+            return []
+
+        S = [0] * size
+        S[0] = 1
+        for k in range(1, size):
+            S[k] = k * S[k - 1]
+        for k in range(1, size - 1):
+            for j in range(k, size):
+                S[j] = (j - k) * S[j - 1] + (j - k + 1) * S[j]
+        return S
```

Comment:
I think if the argument  is a symbol, you should remain it unevaulated rather than returning a empty list i.e return ``None`` in the ``eval`` method.